### PR TITLE
optimize: reduce unnecessary dependences in client side

### PR DESCRIPTION
**Pull request by ggndnn**
### Ⅰ. Describe what this PR did
There are some unnecessary transitive dependencies of client side artifacts, e.g. rm-datasource, that may lead to dependent artifact conflict in user's environment. 

```
...
<dependency>
    <groupId>com.esotericsoftware</groupId>
    <artifactId>kryo</artifactId>
</dependency>
<dependency>
    <groupId>de.javakaffee</groupId>
    <artifactId>kryo-serializers</artifactId>
</dependency>
...
```

This PR tried to remove unnecessary transitive dependencies of client side artifacts from pom.xml as much as possible.

Please have a review to check whether these changes are appropriate.

### CODE REVIEW

**Comment by slievrly** [comment]
if user use seata-all dependency, it will not contains provided scope dependency in all.pom.
image
image

**Comment by ggndnn** [comment]
It's ok to use seata-all as dependency. But if user use rm-datasource directly, some dependencies are unnecessary, which should be optional.

**Comment by zjinlei** [comment]
please resolve conflicts

File: codec/seata-codec-protobuf/pom.xml [hunk 1]
```diff
@@ -37,5 +37,10 @@
             <artifactId>seata-core</artifactId>
             <version>${project.version}</version>
         </dependency>
+
+        <dependency>
```
Comment (slievrly) [inline codec/seata-codec-protobuf/pom.xml]: `codec` module renamed as `serializer`

Comment (ggndnn) [reply codec/seata-codec-protobuf/pom.xml]: fixed

**Comment by ggndnn** [comment]
> please resolve conflicts
fixed

**Pull request merged**